src/components/copyField/copyField.js

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Form, Icon, Button } from 'patternfly-react';
+import helpers from '../../common/helpers';
+
+class CopyField extends React.Component {
+  static copyClipboard(text) {
+    let successful;
+
+    try {
+      window.getSelection().removeAllRanges();
+
+      const newTextarea = document.createElement('textarea');
+      newTextarea.innerHTML = text;
+      newTextarea.style.position = 'absolute';
+      newTextarea.style.top = '-1000px';
+      newTextarea.style.left = '-1000px';
+      newTextarea.style.overflow = 'hidden';
+      newTextarea.style.width = '1px';
+      newTextarea.style.height = '1px';
+
+      const range = document.createRange();
+      window.document.body.appendChild(newTextarea);
+
+      range.selectNode(newTextarea);
+      window.getSelection().addRange(range);
+
+      successful = window.document.execCommand('copy');
+
+      window.document.body.removeChild(newTextarea);
+      window.getSelection().removeAllRanges();
+    } catch (e) {
+      successful = false;
+      console.warn('Copy to clipboard failed.', e.message);
+    }
+
+    return successful;
+  }
+
+  state = {
+    copied: false,
+    expanded: false,
+    timer: null
+  };
+
+  onCopy = event => {
+    const { value } = this.props;
+    const success = CopyField.copyClipboard(value);
+
+    event.target.blur();
+    clearTimeout(this.state.timer);
+
+    this.setState(
+      {
+        copied: success
+      },
+      () => this.resetStateTimer()
+    );
+  };
+
+  onExpand = event => {
+    event.target.blur();
+
+    this.setState({
+      expanded: !this.state.expanded
+    });
+  };
+
+  resetStateTimer() {
+    const timer = setTimeout(
+      () =>
+        this.setState({
+          copied: false
+        }),
+      1500
+    );
+
+    this.setState({ timer });
+  }
+
+  render() {
+    const { copied, expanded } = this.state;
+    const {
+      id,
+      label,
+      labelDescription,
+      multiline,
+      toggleDescription,
+      expandDescription,
+      value,
+      ...props
+    } = this.props;
+    const setId = id || helpers.generateId();
+
+    return (
+      <Form.FormGroup className="cloudmeter-copy" controlId={setId} aria-live="polite" {...props}>
+        <Form.InputGroup>
+          {multiline && (
+            <Form.InputGroup.Button>
+              <Button onClick={this.onExpand} className="cloudmeter-copy-display-button" aria-label={toggleDescription}>
+                {!expanded && <Icon type="fa" name="angle-right" />}
+                {expanded && <Icon type="fa" name="angle-down" />}
+              </Button>
+            </Form.InputGroup.Button>
+          )}
+          <Form.FormControl
+            type="text"
+            value={value}
+            className={`cloudmeter-copy-input ${expanded && 'expanded'}`}
+            disabled
```

Comment:
Requested changes 
- change to `readOnly`
- consider an `on select` click event for contents